fix: lua getNumber overflow with "MoveEvent::EquipItem" function

dudantas opened the pull request:
Fixed overflow with "MoveEvent::EquipItem" and added imbuement shrine to canary temple

dudantas committed:
fix: lua getNumber overflow with "MoveEvent::EquipItem" function

github-actions[bot] committed:
Lua code format - (Stylua)

Regeton5306 commented:
After adding this, it makes it impossible to pick things up from the ground. Anything I drop on the ground can no longer be picked up again.

dudantas commented:
> After adding this, it makes it impossible to pick things up from the ground. Anything I drop on the ground can no longer be picked up again.

Are you sure you tested this correctly? I couldn't reproduce what you said, I tested it and everything is ok. Also, the if I changed only deals with the reward chest/boss corpse, there would be no way to bug other behaviors.

Regeton5306 commented:
> > After adding this, it makes it impossible to pick things up from the ground. Anything I drop on the ground can no longer be picked up again.
> 
> Are you sure you tested this correctly? I couldn't reproduce what you said, I tested it and everything is ok. Also, the if I changed only deals with the reward chest/boss corpse, there would be no way to bug other behaviors.


My base is clean, in fact, I downloaded it about a week ago and I have been adding every update they have merged. I just tried this one and I hadn't noticed until I threw the things I had on the ground and tried to pick them up, but it didn't let me. It just said 'sorry, not possible.' Then I remembered that the only thing I had done was add this, so I reverted the changes, recompiled, and it let me pick them up again. I don't understand.

dudantas commented:
> > > After adding this, it makes it impossible to pick things up from the ground. Anything I drop on the ground can no longer be picked up again.
> > 
> > 
> > Are you sure you tested this correctly? I couldn't reproduce what you said, I tested it and everything is ok. Also, the if I changed only deals with the reward chest/boss corpse, there would be no way to bug other behaviors.
> 
> My base is clean, in fact, I downloaded it about a week ago and I have been adding every update they have merged. I just tried this one and I hadn't noticed until I threw the things I had on the ground and tried to pick them up, but it didn't let me. It just said 'sorry, not possible.' Then I remembered that the only thing I had done was add this, so I reverted the changes, recompiled, and it let me pick them up again. I don't understand.

Maybe it was compilation garbage, the change in the source of this pull request is not even relevant, it does not change the behavior of anything, it just improves the log to identify which function is causing the error.
